kedro_azureml/generator.py

```diff
@@ -172,12 +172,24 @@ def _get_output(self, name):
         if name in self.catalog.list() and isinstance(
             ds := self.catalog._get_dataset(name), AzureMLAssetDataSet
         ):
+            output_path = (
+                f"azureml://datastores/{ds._datastore}/paths/{ds._azureml_root_dir}"
+            )
+
+            # versioning system: to be discussed
+            output_path = (
+                f"{output_path}/{ds._azureml_dataset}/{ds.resolve_save_version()}"
```

Comment:
The main problem I see with this approach is that this hard codes the version in the pipeline specification. This might not be an issue for a single run using `kedro azureml run`, but it means a rerun of that pipeline or a scheduled job created from the result of `kedro azureml compile` will (attempt) to save to the same location.

Ideally, the versioning would be added to the path when the node is run, but that doesn't look feasible given that the directory has to be empty for Azure ML to accept it.

I might be overlooking something, but I now feel that we won't be able to add a sensible implementation of this feature unless Azure ML accepts non-empty output directories...